src/internal/modules/invoice-accounts/controllers/invoice-accounts.js

```diff
@@ -45,20 +46,46 @@ const postCompany = async (request, h) => {
   const form = forms.handleRequest(selectCompanyForm(request, company), request, schema);
   if (form.isValid) {
     const redirectPath = helpers.processCompanyFormData(request, regionId, companyId, form);
-    return h.redirect(`/invoice-accounts/create/${regionId}/${companyId}/${redirectPath}`);
+    return h.redirect(redirectPath);
   }
-  const { viewData: { redirectPath } } = dataService.sessionManager(request, regionId, companyId);
-  return h.postRedirectGet(form, urlJoin('/invoice-accounts/create/', regionId, companyId), { redirectPath });
+  const { viewData } = dataService.sessionManager(request, regionId, companyId);
+  return h.postRedirectGet(form, urlJoin('/invoice-accounts/create/', regionId, companyId), { redirectPath: viewData.redirectPath });
+};
+
+// Handles the redirection to the address entry flow
+const getCreateAddress = async (request, h) => {
+  const { regionId, companyId } = request.params;
+  const queryTail = queryString.stringify({
+    redirectPath: `/invoice-accounts/create/${regionId}/${companyId}/address-entered`,
+    back: `/invoice-accounts/create/${regionId}/${companyId}/check-details`
+  });
+  return h.redirect(`/address-entry/postcode?${queryTail}`);
+};
+
+// Handles the address entered via the address entry flow, and stores it in the session object
+const getAddressEntered = async (request, h) => {
+  const { regionId, companyId } = request.params;
+  // Fetch the address using the address flow session key
+  const address = await sessionHelper.saveToSession(request, ADDRESS_FLOW_SESSION_KEY);
+  // Store the address in the session object
+  dataService.sessionManager(request, regionId, companyId, {
+    address: {
+      ...address,
+      id: address.id ? address.id : tempId
+    }
+  });
+  // Redirect the user to the check your answers page.
+  return h.redirect(`/invoice-accounts/create/${regionId}/${companyId}/check-details`);
 };
 
 const getAddress = async (request, h) => {
   const { regionId, companyId } = request.params;
-  const addresses = await dataService.getCompanyAddresses(companyId);
   // get the session data to check if the address has been set and if it is new or existing
-  const session = dataService.sessionManager(request, regionId, companyId);
+  const session = await dataService.sessionManager(request, regionId, companyId);
```

Comment:
To accommodate the redirect back from the contact entry plugin this line should change to:
```
  // get the session data to check if the address has been set and if it is new or existing
  const session = await dataService.sessionManager(request, regionId, companyId, { agent: agentId });
```
The next line of code where it gets the addresses have already been written to pull in the addresses for an agent id if it is present.